src/reducers/selectors.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+
+import { createSelector } from 'reselect';
+
+export const EMPTY_OBJ = {};
+export const EMPTY_AR = [];
+
+// Checks a response object for errors and returns them
+export const processErrors = (responseValue: ?QueryResponse): Object => {
+	if (!responseValue || !responseValue.error) {
+		return {};
+	}
+	return responseValue;
+};
+
+export const processErrorsMaybe = (responseValue: ?QueryResponse) => {
+	const processed = processErrors(responseValue);
+	return Object.keys(processed).length ? processed : null;
+};
+
+// Checks if object is empty
+export const isEmpty: Selector<boolean> = state => {
+	if (!state) {
+		return true;
+	}
+	return Object.keys(state).length === 0;
+};
+
+// Checks if object is has errors
+export const hasErrors: Selector<boolean> = state =>
+	Object.keys(processErrors(state)).length > 0;
+
+// Combo check if the object is either empty or has errors
+export const hasValidValue: Selector<boolean> = state =>
+	!isEmpty(state) && !hasErrors(state);
+
+/*
+ * returns the QueryResponse.value, or an object containing a single `error`
+ * property. An empty response returns the defaultValue.
+ */
+export const getValue = (
+	resp: ?QueryResponse,
+	defaultValue: mixed = EMPTY_OBJ
+): mixed => {
+	if (!resp || isEmpty(resp)) {
```

Comment:
`isEmpty` checks for more than just `!resp`, but I had to explicitly add `!resp` here because Flow wasn't smart enough to know that `isEmpty` handles the null case, unfortunately. A little more verbose, but I've convinced myself that is also is more obvious to someone reading the code without being too verbose, which is a good thing.

yeah - the expectation is that if you are using looking for an array value, you will need to provide the `defaultValue`. I wouldn't mind making the `defaultValue` a required argument, but that would be a followup ticket with a breaking change.